qa/src/test/java/io/camunda/migrator/qa/CustomJobTypeValidationTest.java

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Camunda License 1.0. You may not use this file
+ * except in compliance with the Camunda License 1.0.
+ */
+
+package io.camunda.migrator.qa;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import io.camunda.migrator.RuntimeMigrator;
+import io.camunda.migrator.qa.util.RuntimeMigrationAbstractTest;
+import io.github.netmikey.logunit.api.LogCapturer;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.RegisterExtension;
+import org.springframework.test.context.TestPropertySource;
+
+@TestPropertySource(properties = {
+    "camunda.migrator.job-type=custom-job-type"
+})
+public class CustomJobTypeValidationTest extends RuntimeMigrationAbstractTest {
+
+  @RegisterExtension
+  protected LogCapturer logs = LogCapturer.create().captureForType(RuntimeMigrator.class);
+
+  @Test
+  public void shouldUseCustomJobTypeInValidationMessage() {
+    // given
+    deployProcessInC7AndC8("noMigratorListener.bpmn");
+
+    String id = runtimeService.startProcessInstanceByKey("noMigratorListener").getId();
+
+    // assume
+    assertThat(runtimeService.createProcessInstanceQuery().processInstanceId(id).singleResult()).isNotNull();
+
+    // when
+    runtimeMigrator.start();
+
+    // then
+    assertThatProcessInstanceCountIsEqualTo(0);
+
+    var events = logs.getEvents();
+    assertThat(events.stream()
+        .filter(event -> event.getMessage()
+            .matches(String.format(".*Skipping process instance with legacyId \\[%s\\]: "
+                + "Couldn't find execution listener of type 'custom-job-type' on start event "
+                + "\\[Event_1px2j50\\] in C8 process with key \\[(\\d+)\\]\\.", id))))
+        .hasSize(1);
+  }
+
+  @Test
+  public void shouldUseCustomJobTypeInListenerNotFoundMessage() {
+    // given
+    deployProcessInC7AndC8("migratorListenerCustomType.bpmn");
+
+    String id = runtimeService.startProcessInstanceByKey("migratorListenerCustomType").getId();
```

Comment:
💭 I did not immediately understand this test case because it looked so similar to the above. Looking at the models I understand: case 1 is that there is no listener at all, and case 2 is that there is a listener, but its got the wrong type right?

Maybe we can adjust model names to make this more obvious:
```suggestion
    deployProcessInC7AndC8("migratorListenerWrongType.bpmn");

    String id = runtimeService.startProcessInstanceByKey("migratorListenerWrongType").getId();
```